update inputs.modbus serial module dependency to unbreak OpenBSD

### Relevent telegraf.conf

```toml
Well, not reaching config step... right now openbsd ships telegraf 1.13.4, which is old.
```


### System info

Telegraf 1.20.3 on OpenBSD 7.0

### Docker

_No response_

### Steps to reproduce

1. Try to build telegraf last releases on OpenBSD
2. See it badly fail, cf goburrow/serial#18
...


### Expected behavior

Build should succeed

### Actual behavior

It fails, as serial module got support for 'bsd' in goburrow/serial@a9d73190f6 but afaict this was only working on FreeBSD.

### Additional info

_No response_

Note that telegraf also depends on grid-x/serial which is also broken on OpenBSD, thus telegraf will still fail to build:
```
github.com/grid-x/serial                                                                                  
# github.com/grid-x/serial                                                                                
../../../go/pkg/mod/github.com/grid-x/serial@v0.0.0-20191104121038-e24bc9bf6f08/serial_bsd.go:30:10: undefined: syscall.B460800                                                                                      
../../../go/pkg/mod/github.com/grid-x/serial@v0.0.0-20191104121038-e24bc9bf6f08/serial_bsd.go:78:44: fds.X__fds_bits undefined (type *syscall.FdSet has no field or method X__fds_bits)                             
../../../go/pkg/mod/github.com/grid-x/serial@v0.0.0-20191104121038-e24bc9bf6f08/serial_bsd.go:79:45: fds.X__fds_bits undefined (type *syscall.FdSet has no field or method X__fds_bits)                              
../../../go/pkg/mod/github.com/grid-x/serial@v0.0.0-20191104121038-e24bc9bf6f08/serial_bsd.go:86:5: fds.X__fds_bits undefined (type *syscall.FdSet has no field or method X__fds_bits)                              
../../../go/pkg/mod/github.com/grid-x/serial@v0.0.0-20191104121038-e24bc9bf6f08/serial_bsd.go:92:12: fds.X__fds_bits undefined (type *syscall.FdSet has no field or method X__fds_bits)                              
../../../go/pkg/mod/github.com/grid-x/serial@v0.0.0-20191104121038-e24bc9bf6f08/termios_bsd.go:10:17: cannot use speed (type uint32) as type int32 in assignment                                                     
../../../go/pkg/mod/github.com/grid-x/serial@v0.0.0-20191104121038-e24bc9bf6f08/termios_bsd.go:14:17: cannot use speed (type uint32) as type int32 in assignment
```

--- a/plugins/inputs/modbus/configuration.go
+++ b/plugins/inputs/modbus/configuration.go
@@ -1,3 +1,5 @@
+//go:build !openbsd
+
 package modbus
 
 import "fmt"

--- a/plugins/inputs/modbus/configuration_original.go
+++ b/plugins/inputs/modbus/configuration_original.go
@@ -1,3 +1,5 @@
+//go:build !openbsd
+
 package modbus
 
 import (

--- a/plugins/inputs/modbus/modbus.go
+++ b/plugins/inputs/modbus/modbus.go
@@ -1,3 +1,5 @@
+//go:build !openbsd
+
 package modbus
 
 import (

--- a/plugins/inputs/modbus/modbus_openbsd.go
+++ b/plugins/inputs/modbus/modbus_openbsd.go
@@ -1,0 +1,3 @@
+//go:build openbsd
+
+package modbus

--- a/plugins/inputs/modbus/modbus_test.go
+++ b/plugins/inputs/modbus/modbus_test.go
@@ -1,3 +1,5 @@
+//go:build !openbsd
+
 package modbus
 
 import (

--- a/plugins/inputs/modbus/request.go
+++ b/plugins/inputs/modbus/request.go
@@ -1,3 +1,5 @@
+//go:build !openbsd
+
 package modbus
 
 import "sort"

--- a/plugins/inputs/modbus/type_conversions.go
+++ b/plugins/inputs/modbus/type_conversions.go
@@ -1,3 +1,5 @@
+//go:build !openbsd
+
 package modbus
 
 import "fmt"

--- a/plugins/inputs/modbus/type_conversions16.go
+++ b/plugins/inputs/modbus/type_conversions16.go
@@ -1,3 +1,5 @@
+//go:build !openbsd
+
 package modbus
 
 import (

--- a/plugins/inputs/modbus/type_conversions32.go
+++ b/plugins/inputs/modbus/type_conversions32.go
@@ -1,3 +1,5 @@
+//go:build !openbsd
+
 package modbus
 
 import (

--- a/plugins/inputs/modbus/type_conversions64.go
+++ b/plugins/inputs/modbus/type_conversions64.go
@@ -1,3 +1,5 @@
+//go:build !openbsd
+
 package modbus
 
 import (

